homeassistant/components/withings/common.py

```diff
@@ -280,58 +655,393 @@ async def update_sleep_summary(self) -> SleepGetSummaryResponse:
+        bad params, etc. Simply retrying the call is enough to workaround these issues.
+        """
+        last_exception = None
+        max_attempts = 3
+        for attempt in range(1, max_attempts + 1):
+            _LOGGER.debug(
+                "Attempt %s/%s attempting call %s.",
+                attempt,
+                max_attempts,
+                str(function),
+            )
+            try:
+                result = await self._hass.async_add_executor_job(function)
+                self._on_api_success()
+                return result
+            except Exception as ex:  # pylint: disable=broad-except
+                last_exception = ex
+                if attempt < max_attempts:
+                    await asyncio.sleep(0.3)
+
+        final_exception = None
+
+        auth_exceptions = (UnauthorizedException, AuthFailedException)
+        if final_exception is None and isinstance(last_exception, auth_exceptions):
+            final_exception = NotAuthenticatedError(last_exception)
+
+        warning_exceptions = (
+            TooManyRequestsException,
+            InvalidParamsException,
+            ErrorOccurredException,
+            TimeoutException,
+            BadStateException,
+            UnknownStatusException,
+        )
+        if final_exception is None and isinstance(last_exception, warning_exceptions):
+            _LOGGER.exception("Withings exception occurred.")
+            final_exception = last_exception
+
+        # Oauth2 config flow failed to authenticate.
+        if final_exception is None and NOT_AUTHENTICATED_ERROR.match(
+            str(last_exception)
+        ):
+            final_exception = NotAuthenticatedError(last_exception)
+        elif final_exception is None:
+            # Probably a network error.
+            final_exception = PlatformNotReady(last_exception)
+
+        await self._async_on_api_error(final_exception)
+        raise final_exception
+
+    async def async_webhook_data_updated(self, data_category: NotifyAppli) -> None:
+        """Handle scenario when data is updated from a webook."""
+        _LOGGER.info("Withings webhook triggered.")
+        print("BBBBBBB")
+        if data_category in {
+            NotifyAppli.WEIGHT,
+            NotifyAppli.CIRCULATORY,
+        }:
+            await self.async_update_measure()
+
+        if data_category in {NotifyAppli.SLEEP}:
+            await self.async_update_sleep_summary()
+
+        if data_category in {NotifyAppli.BED_IN, NotifyAppli.BED_OUT}:
+            await self.async_set_data(
+                WITHINGS_MEASUREMENTS_MAP[Measurement.IN_BED],
+                data_category == NotifyAppli.BED_IN,
+            )
+
+
+def get_attribute_unique_id(
+    attribute: WithingsAttribute, profile: str, user_id: int
+) -> str:
+    """Get a entity unique id for a user's attribute."""
+    measurement_slug = slugify(attribute.measurement.value)
+    return f"withings_{slugify(profile)}_{user_id}_" f"{measurement_slug}"
```

Comment:
Why slugify? 

Also, you are using two strings but can use 1
```suggestion
    measurement_slug = slugify(attribute.measurement.value)
    return f"withings_{slugify(profile)}_{user_id}_{measurement_slug}"
```

Reply:
This needs to look like a unique_id. The profile could contains spaces, punctuation and various cases. So slugify will help keep this consistent.

Reply:
> needs to look like a unique ID

I am not aware of any such requirement.

Also, is profile something that can be changed by a user ? If so, why not just use user_id and measurement ?